### DISCUSSION TITLE
Is it possible to make multi-step forms?

### DISCUSSION THREAD

**Discussion by jkinley** [TanStack/form#843] [Q&A] [answered]
Just wondering. I have looked through the docs and don't see any examples.

**Comment by Balastrong** [comment, marked as answer]
It is possible but with some manual work. We don't have (yet) a dedicated API but it is indeed a feature we'd like to support after v1.x
See #419

**Reply by noahkurz** [reply]
Any rough estimate on when v1 will be released? Super excited for it!

**Reply by AndrejNemec** [reply]
+1

**Reply by divya-nsh** [reply]
> It is possible but with some manual work. We don't have (yet) a dedicated API but it is indeed a feature we'd like to support after v1.x See #419

Can you share any example (using ZOD validator)

**Reply by peterwiebe** [reply]
With some type coercion I was able to get it to work, in my scenario each step is its own component which is not as clean as I would have liked it to be. Passing the form object as a prop is not pleasant

**Reply by DiegoJohnsonL** [reply]
It is not pleasant at all. i had not idea how to infer the types for my form attributes so i first put everything into a single form but then validating each value per step was a nightmare so im about to refactor to do a form per step which isn't ideal at all
